### PR TITLE
Make os.stat_result and friends NamedTuples

### PR DESCRIPTION
**Pull request by euresti**
The elements can be accessed as indexes or through attributes, so I chose to make them NamedTuples.

### CODE REVIEW

File: stdlib/2/os/__init__.pyi [hunk 1]
```diff
@@ -94,10 +95,58 @@ WUNTRACED = 0  # Unix only
 
 TMP_MAX = 0  # Undocumented, but used by tempfile
 _PathType = Union[bytes, Text]
-_StatVFS = NamedTuple('_StatVFS', [('f_bsize', int), ('f_frsize', int), ('f_blocks', int),
-                                   ('f_bfree', int), ('f_bavail', int), ('f_files', int),
-                                   ('f_ffree', int), ('f_favail', int), ('f_flag', int),
-                                   ('f_namemax', int)])
+
+class stat_result(NamedTuple('stat_result', [
+        ('st_mode', int),  # protection bits,
```
Comment (JelleZijlstra) [inline stdlib/2/os/__init__.pyi]: I don't think we need all of those comments. Typeshed is for types, not documentation.

File: stdlib/2/os/__init__.pyi [hunk 2]
```diff
@@ -94,10 +95,58 @@ WUNTRACED = 0  # Unix only
...
+        ('st_ctime', float)])):  # platform dependent (time of most recent metadata change on Unix, or the time of creation on Windows)
+
+    # For backward compatibility, the return value of stat() is also
+    # accessible as a tuple of at least 10 integers giving the most important
+    # (and portable) members of the stat structure, in the order st_mode,
+    # st_ino, st_dev, st_nlink, st_uid, st_gid, st_size, st_atime, st_mtime,
+    # st_ctime. More items may be added at the end by some implementations.
+
+    if sys.version_info >= (3, 3):
+        st_atime_ns = 0  # time of most recent access, in nanoseconds
```
Comment (JelleZijlstra) [inline stdlib/2/os/__init__.pyi]: Better to write `st_atime_ns: int` (or `st_atime_ns = ...  # type: int`, but I think we've established that all users of typeshed obey PEP 526 now).

File: stdlib/3/os/__init__.pyi [hunk 3]
```diff
@@ -325,8 +321,11 @@ if sys.version_info >= (3, 5):
     def scandir(path: str = ...) -> Iterator[DirEntry[str]]: ...
     @overload
     def scandir(path: bytes) -> Iterator[DirEntry[bytes]]: ...
+@overload
+def stat_float_times(newvalue: bool = ...) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/3/os/__init__.pyi]: I don't think this should have a default value; that will make it overlap with the other overload.

**Comment by JelleZijlstra** [comment]
Thanks! Merged.

**Comment by gvanrossum** [comment]
Hmm... This caused two types of breakage with our internal code. I think it's better to roll back. @JelleZijlstra, @euresti What do you guys think? Here are the errors (same as I reported for #1111):

This one's caused by the (newly introduced) difference between os.stat_result and posix.stat_result:
```
Argument 1 to "check_perm" has incompatible type "stat_result"; expected "posix.stat_result"
```

This one's triggered by `os.stat(path)[stats.ST_MODE]`:
```
Tuple index must be an integer literal
```

Who agrees with a rollback?

**Comment by JelleZijlstra** [comment]
We can fix the first one pretty easily by making stdlib/2/posix.pyi re-export os.stat_result (the one in stdlib/3 already does that).

I don't think the second one can be fixed in the stubs, but maybe mypy is being too strict there (it could just allow that code and make it return Any).

I'm OK with reverting for now, but it would be good to get some form of this PR back in eventually so `os.stat` doesn't just return Any.

**Comment by gvanrossum** [comment]
I filed https://github.com/python/mypy/issues/3078 for the latter. I'll revert if you don't beat me to it.

**Comment by euresti** [comment]
Ack! Sorry about the mess. Please revert and I'll take a look later.

**Pull request merged**